### ISSUE TITLE
Programmatically incorrect role 'generic" is provided for "Error" control.: A11y_TypeSpec_Playground_Name Role Value

### ISSUE THREAD

**Issue by kupatkar99** [microsoft/typespec#8559]
### GIT Hub Tags:
#A11yTCS; #A11ySev2; #TypeSpec; #P2_TypeSpec_Web_Sept2025; #CELA_P2; #DesktopWeb; #A11yMAS; #WCAG4.1.2; #Name Role Value; #AILimited; #Win11; #FTP; #ChromiumEdge; #Developer Tool;

### Environment and OS details:
Application name: typespec.io
URL: https://[typespec.io](https://typespec.io/)/
ChromiumEdge Version: 140.0.3485.66 (Official build) (64-bit)

### Reproduction Steps:

1. Open url "https://[typespec.io](https://typespec.io/playground/?sample=API+versioning)/playground/?sample=API+versioning" and login with valid Credentials.
2. "TypeSpec Playground" Page will get open.
3. Edit the Code with some errors.
4. TAB till Error button in bottom of the editor Error Section.
5. Observe the issue whether Correct role property is provided to "Error" button or not.

### Actual behavior:
 Programmatically incorrect role 'generic" is provided for "Error" control.

### Expected behavior:
Programmatically suitable role "button" Should be provided for "Error" Control.

### User Impact:
Screen Reader users will not be able to get information about the control is role is not programmatically defined.

### Attachment:

<img width="1909" height="1044" alt="Image" src="https://github.com/user-attachments/assets/7d826724-6001-4788-8b3a-c773cfb9a627" />

**Comment by ccbarragan** [comment]
@timotheeguerin Are we using a [MessageBar component](https://storybooks.fluentui.dev/react/?path=/docs/components-messagebar--docs) to display errors in Playground, or maybe a similar component? Suggest using MessageBar as it has built-in roles already defined. Otherwise looks like the fix is already suggested above ^^ (ensure 'button' is added as role instead of 'generic')

**Comment by timotheeguerin** [comment]
@ccbarragan we do not, this is not really a notification error but active info of what is showing in the playground. 
I believe here the problem is only the open trigger is missing the right role which the message bar wouldn't solve anyway.

**Comment by kupatkar99** [comment]
Verified the issue in below environment:
https://[typespec.io](https://typespec.io/playground/?sample=API+versioning)/playground/?sample=API+versioning
issue is fixed. hence closing the bug. please find added attachment for reference.
#Closed;

<img width="1919" height="949" alt="Image" src="https://github.com/user-attachments/assets/e063ed22-0b18-4db6-bff4-883b4e0c98b8" />
